pkg/flink/internal/controller/input_controller.go

```diff
@@ -31,7 +34,28 @@ type InputController struct {
 
 const defaultWindowSize = 100
 
-func NewInputController(history *history.History, lspCompleter prompt.Completer) types.InputControllerInterface {
+func NewLSPHandler(c types.InputControllerInterface, handlerCh chan *jsonrpc2.Request) {
+	if handlerCh == nil {
+		return
+	}
+
+	go func() {
+		for req := range handlerCh {
```

Comment:
oh I didn't know you could do that, so it ranges until the channel is closed?

Reply:
What chatgpt taught me: 

1 - 
```
go func() {
    for {
        select {
            case request := <-reqCh:
                // process request
            default:
                // nothing to do, loop again
        }
    }
}()
```

2 - 
```
go func() {
    for request := range reqCh {
        // process request
    }
}()
```

In 1, the use of the default case in this way can cause high CPU usage because when there are no messages, it will constantly check for them in a tight loop. A better approach might be to remove the default case and let the goroutine block until there's something to receive like in 2.

Reply:
I added one test and also tested it manually with fake mock data and it works ✅

Reply:
yeah the for pattern I knew:
```
for {
        select {
            case request := <-reqCh:
                // process request
        }
    }
```
but not the range one. That's pretty cool